virtual_rainforest/core/config.py

```diff
@@ -1,10 +1,42 @@
-"""The `core.config` module.
-
-The `core.config` module is used to read in the various configuration files, validate
-their contents, and then configure a ready to run instance of the virtual rainforest
-model.
+"""API documentation for the :mod:`core.config` module.
+************************************************** # noqa: D205
+
+The :mod:`core.config` module is used to read in the various configuration files,
+validate their contents, and then configure a ready to run instance of the virtual
+rainforest model. The basic details of how this system is used can be found
+:ref:`here<virtual_rainforest/core/config:the configuration module>`.
+
+When a new module is defined a ``JSON`` file should be written, which includes the
+expected configuration tags, their expected types, any value constraints, and where
+appropriate default values. This schema should be saved in the folder of the module that
+it relates to. In order to make this schema generally accessible to the ``vr`` package,
+it should then be added to the schema registry.  The
+:func:`~virtual_rainforest.core.config.register_schema` decorator is used for this
+purpose.
+
+Schema registration for each module takes place in the module's ``__init__.py``. Here, a
+function is written that reads in the ``JSON`` file describing the schema. This function
+is then decorated using :func:`~virtual_rainforest.core.config.register_schema`, which
+results in the schema being added to the registry. The user provides a schema name to
+the decorator to register the schema under, this should be unique, and generally should
+just be the module name. An example of decorator usage is shown below:
+
+.. code-block:: python
+
+    @register_schema("example_module")
+    def schema() -> dict:
+        schema_file = Path(__file__).parent.resolve() / "example_module_schema.json"
+
+        with schema_file.open() as f:
+            config_schema = json.load(f)
+
+        return config_schema
+
+It's important to note that the schema will only be added to the registry if the module
+``__init__`` is run. This means that somewhere in your path the module must be imported.
```

Comment:
From just reading this section, it is not clear to me what 'the path' is. Is it pointing to the location of the vr package or the module?

Reply:
So basically so long as the module of interest (e.g. `virtual_rainforest.soil`) is imported somewhere in setting up your script to run the registry gets populated. This doesn't have to be direct and you can get away with importing something that imports something else which imports the `virtual_rainforest.soil` module. In this case, the registry would have been properly populated with the soil model.

Path is definitely not a good way to describe this though, I'm not sure how to phrase it. Does "chain of imports" make more sense, or should I aim for a more wordy description of the problem?